x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/CrossClusterAccessAuthenticationService.java

```diff
@@ -108,6 +111,28 @@ public void authenticate(final String action, final TransportRequest request, fi
         }
     }
 
+    public void tryAuthenticateCredentialsHeaderOnly(
+        ThreadContext threadContext,
```

Comment:
At this stage, I think it would help with readability to avoid passing in the `ThreadContext` as a parameter (inside `ApiKeyService#tryAuthenticate` you can use `client.getThreadPool().getThreadContext()`).
I think this avoids the confusion about whether "tryAuthenticate" looks into the thread context or the passed in headers to run the authentication (it only looks at the passed-in headers).